### DISCUSSION TITLE
x-trap does not clear if used on inside template

### DISCUSSION THREAD

**Discussion by magnusskare** [alpinejs/alpine#2673] [7. Bugs (archived)]
As the title states, x-trap does not clear if x-trap is used within a `<template>` tag. ~This worked when x-trap was a stand-alone plugin, before it was renamed to focus.~

### Expected behavior
The set overflow should be cleared, along with aria-hidden attributes.

### Actual behavior
x-data "search" is set to false, but logic related to "search" inside the `<template>` tag is not evaluated.

### Test
Reduced test-case: https://codepen.io/partcoffee/pen/ZEaXrao
Clicking "Trigger search" will show some HTML inside a `<template>` tag, using x-if. Hiding said template hides the HTML, but overflow and aria-attributes are not removed from the document, making tabbing and focusing impossible.

**Comment by SimoTod** [comment]
I believe it's because of https://github.com/alpinejs/alpine/pull/2556/files
v3.8.1 should work as a temporary workaround.

**Reply by magnusskare** [reply]
Good catch, you're right — this works as a temporary workaround. Thank you!

**Reply by SimoTod** [reply]
Created https://github.com/alpinejs/alpine/pull/2682 to fix the bug.
